### DISCUSSION TITLE
Cache regeneration lock

### DISCUSSION THREAD

**Discussion by elidrissidev** [OpenMage/magento-lts#2299] [Ideas]
We have been having this issue with cache regeneration taking our site down when there's high traffic for a long time, pretty sure it happens with other things like db scripts and maybe indexing too?

I came across one related extension today that modifies `Mage_Core_Model_App` and it seems the fix is quite short and easy to implement, so I wonder if it will be a good idea to implement it, at least in v20.

**Comment by addison74** [comment]
The existence of two versions starting to be an issue. We'll see how long it goes this way. My opinion that OM-20 should be the default branch from now on and only certain PRs of it should be taken to OM-19. Thus we focus more on an optimized solution than on one where we have to take care of trinkets.

Let it go directly in 19 and we'll see later if it's only for 20. The maintainers helped me in the past and took PR where it should.

**Comment by fballiano** [comment]
I wrote a patch for that many years ago, it’s somewhere in my github 😅

**Reply by fballiano** [reply]
https://github.com/fballiano/magento-cache-regeneration-lock
